src/app/components/animate/animate.ts

```diff
@@ -0,0 +1,178 @@
+            rect.bottom <= ((window.innerHeight + rect.height) || this.document.documentElement.clientHeight) &&
+            rect.right <= (window.innerWidth || this.document.documentElement.clientWidth)
+        );
+    }
+
+    isImage(): boolean {
+        return this.el.nativeElement.tagName === 'IMG';
+    }
+
+    animate() {
+        if (this.loaded) {
+            if (this.isInViewport() && !this.entered) {
+                this.enter();
+            }
+
+            if (this.isInViewport() && this.entered) {
+                this.enter();
+            }
+
+            if (!this.isInViewport() && !this.entered) {
+                this.leave();
+            }
+        }
+    }
+
+    enter() {
+        this.host.nativeElement.style.visibility = 'visible';
+        DomHandler.addClass(this.host.nativeElement, this.enterClass);
+    }
+
+    leave() {
+        this.host.nativeElement.style.visibility = 'hidden';
+        if (this.leaveClass) {
+            DomHandler.addClass(this.host.nativeElement, this.leaveClass);
+        }
+        DomHandler.removeClass(this.host.nativeElement, this.enterClass);
+    }
+
+    blockOverflow() {
+        if (this.parentElement.nativeElement) {
+            DomHandler.addClass(this.parentElement.nativeElement, 'overflow-x-hidden');
+        }
+        if (!this.parentElement.nativeElement && this.parentElement) {
+            DomHandler.addClass(this.parentElement, 'overflow-x-hidden');
+        }
+    }
+
+    unblockOverflow() {
+        if (this.parentElement.nativeElement) {
+            DomHandler.removeClass(this.parentElement.nativeElement, 'overflow-x-hidden');
+        }
+        if (!this.parentElement.nativeElement && this.parentElement) {
+            DomHandler.removeClass(this.parentElement, 'overflow-x-hidden');
+        }
+    }
+
+    bindDocumentScrollListener() {
+        if (!this.documentScrollListener) {
+            this.documentScrollListener = this.renderer.listen(window, 'scroll', () => {
+                this.scroll = true;
+                if (!this.observer) {
+                    this.constructIntersectionObserver();
+                }
+                this.animate();
+            })
+        }
+    }
+
+    unbindDocumentScrollListener() {
+        if (this.documentScrollListener) {
+            this.scroll = false;
+            this.documentScrollListener();
+            this.documentScrollListener = null;
+        }
+    }
+
+    bindLoadListener() {
+        this.loadListener = this.renderer.listen(this.el.nativeElement, 'load', () => {
+            if (!this.loaded) {
+                this.animate();
+            }
+            if (!this.documentScrollListener) {
+                this.bindDocumentScrollListener();
+            }
+            this.loaded = true;
+        });
+    }
+
+    unbindLoadListener() {
+        if (this.loadListener) {
+            this.loadListener();
+            this.loadListener = null;
+        }
+    }
+
+    ngOnDestroy() {
```

Comment:
Hi, you also need to unsubscribe from IntersecrtionObserver(use method unobserve)